src/v/debug_bundle/debug_bundle_service.cc

```diff
@@ -87,6 +95,48 @@ bool is_valid_k8s_namespace(std::string_view ns) {
     constexpr auto max_ns_length = 63;
     return !ns.empty() && ns.size() <= max_ns_length && is_valid_rfc1123(ns);
 }
+
+std::filesystem::path form_debug_bundle_storage_directory() {
+    const auto& debug_bundle_dir
+      = config::shard_local_cfg().debug_bundle_storage_dir();
+
+    // Either return the storage directory or the data directory appended with
+    // "debug-bundle"
+    return debug_bundle_dir.value_or(
+      config::node().data_directory.value().path
+      / service::debug_bundle_dir_name);
+}
+
+ss::future<bytes> calculate_sha256_sum(std::string_view path) {
+    const size_t buffer_size = 8192;
+    crypto::digest_ctx ctx(crypto::digest_type::SHA256);
+    auto handle = co_await ss::open_file_dma(path, ss::open_flags::ro);
+    auto h = ss::defer(
+      [handle]() mutable { ssx::background = handle.close(); });
+
+    auto stream = ss::make_file_input_stream(
+      handle,
+      ss::file_input_stream_options{
+        .buffer_size = buffer_size, .read_ahead = 1});
+
+    auto file_size = co_await handle.size();
+    size_t total_consumed = 0;
+
+    while (total_consumed < file_size) {
+        auto buf = co_await stream.read_up_to(
+          std::min(buffer_size, file_size - total_consumed));
+        total_consumed += buf.size();
+        ctx.update({buf.get(), buf.size()});
```

Comment:
You can use stream::read to return whatever is available, without having to choose a specific buffer size.

```cpp
auto buf = co_await stream.read();
if (buf.empty()) {
  break;
}
ctx.update(buf.get(), buf.size());
```